File: feature/wanandroid/pkg/src/main/java/com/open9527/wanandroid/pkg/cell/ShareContentTextCell.java
package com.open9527.wanandroid.pkg.cell;

import android.text.TextUtils;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.databinding.ObservableField;

import com.android.open9527.recycleview.adapter.BaseBindingCell;
import com.android.open9527.recycleview.adapter.BaseBindingCellViewHolder;
import com.blankj.utilcode.util.ToastUtils;
import com.open9527.wanandroid.pkg.BR;
import com.open9527.wanandroid.pkg.R;
import com.open9527.wanandroid.pkg.main.h5.H5Activity;
import com.open9527.wanandroid.pkg.net.ContentVo;

/**
 * @author open_9527
 * Create at 2021/1/14
 **/
public class ShareContentTextCell extends BaseBindingCell<ShareContentTextCell> {

    public final ObservableField<String> valueTitle = new ObservableField<>();
    private final ObservableField<String> valueLink = new ObservableField<>();
    public final ObservableField<String> valueShareUser = new ObservableField<>();
    public final ObservableField<String> valueNiceShareDate = new ObservableField<>();


    public ShareContentTextCell(ContentVo contentVo) {
        super(R.layout.share_content_text_cell);
        valueTitle.set(contentVo.getTitle());
        valueLink.set(contentVo.getLink());

        if (TextUtils.isEmpty(contentVo.getShareUser())) {
            valueShareUser.set("作者:" + contentVo.getAuthor());
        } else {
            valueShareUser.set("分享人:" + contentVo.getShareUser());
        }

        valueNiceShareDate.set(contentVo.getNiceShareDate());
    }

    @Override
    public void bind(@NonNull BaseBindingCellViewHolder holder, int position) {
        holder.addBindingParam(BR.cell, this);
    }

    @Override
    public void onCellClick(View view, ShareContentTextCell shareContentTextCell) {

        H5Activity.startH5(shareContentTextCell.valueLink.get(),shareContentTextCell.valueTitle.get());
    }
}

File: feature/wanandroid/pkg/src/main/java/com/open9527/wanandroid/pkg/cell/ShareContentTextCell.kt
package com.open9527.wanandroid.pkg.cell

import android.text.TextUtils
import android.view.View
import androidx.databinding.ObservableField
import com.android.feature.webview.export.api.WebApi
import com.android.feature.webview.export.api.WebBundle
import com.android.open9527.recycleview.adapter.BaseBindingCell
import com.android.open9527.recycleview.adapter.BaseBindingCellViewHolder
import com.blankj.utilcode.util.ApiUtils
import com.blankj.utilcode.util.StringUtils
import com.open9527.wanandroid.pkg.BR
import com.open9527.wanandroid.pkg.R
import com.open9527.wanandroid.pkg.main.h5.H5Activity
import com.open9527.wanandroid.pkg.main.h5.H5Activity.Companion.startH5
import com.open9527.wanandroid.pkg.net.ContentVo
import java.util.*

/**
 * @author open_9527
 * Create at 2021/1/14
 */
class ShareContentTextCell(contentVo: ContentVo) :
    BaseBindingCell<ShareContentTextCell>(R.layout.share_content_text_cell) {
    @JvmField
    val valueTitle = ObservableField<String?>()
    private val valueLink = ObservableField<String?>()
    private val valueId =  ObservableField<String?>()
    @JvmField
    val valueShareUser = ObservableField<String>()

    @JvmField
    val valueNiceShareDate = ObservableField<String?>()
    override fun bind(holder: BaseBindingCellViewHolder<*>, position: Int) {
        holder.addBindingParam(BR.cell, this)
    }

    override fun onCellClick(view: View, shareContentTextCell: ShareContentTextCell) {
        shareContentTextCell.valueLink.get()?.let {
            if (ApiUtils.getApi(WebApi::class.java).toString().contains("WebApiMockImpl")) {
                startH5(
                    it,
                    shareContentTextCell.valueTitle.get()
                )
            } else {
                ApiUtils.getApi(WebApi::class.java)
                    .startWeb(WebBundle(it, shareContentTextCell.valueTitle.get()))
            }
        }
    }

    init {
        valueId.set(contentVo.id)
        valueTitle.set(contentVo.title)
        valueLink.set(contentVo.link)
        if (TextUtils.isEmpty(contentVo.shareUser)) {
            valueShareUser.set(
                String.format(
                    StringUtils.getString(R.string.wan_share_author),
                    contentVo.author
                )
            )
        } else {
            valueShareUser.set(
                String.format(
                    StringUtils.getString(R.string.wan_share_sharer),
                    contentVo.shareUser
                )
            )
        }
        valueNiceShareDate.set(contentVo.niceShareDate)
    }

    override fun getUUID(): String? {
        return valueId.get()
    }
}